File: frontend/src/types/job.ts
export type JobStatus =
  | 'planned'
  | 'in_progress'
  | 'waiting_parts'
  | 'completed'
  | 'on_hold'

export type JobItem = {
  id: number
  jobCode: string
  title: string
  customerName: string
  assignee: string
  startDate: string
  dueDate: string
  status: JobStatus
}

export const STATUS_LABEL: Record<JobStatus, string> = {
  planned: '計画中',
  in_progress: '進行中',
  waiting_parts: '部材待ち',
  completed: '完了',
  on_hold: '保留',
}

File: src/yaguchi_production_system/api/schemas/job.py
"""Job request/response schemas."""

from datetime import date, datetime

from pydantic import BaseModel, ConfigDict, Field

from yaguchi_production_system.models.job import JobStatus


class JobBase(BaseModel):
    """Shared job fields."""

    job_code: str = Field(min_length=1, max_length=50)
    title: str = Field(min_length=1, max_length=255)
    customer_id: int | None = None
    start_date: date | None = None
    due_date: date | None = None
    status: JobStatus = JobStatus.PLANNED
    notes: str | None = None


class JobCreate(JobBase):
    """Payload for creating jobs."""


class JobUpdate(BaseModel):
    """Payload for updating jobs."""

    title: str | None = Field(default=None, min_length=1, max_length=255)
    customer_id: int | None = None
    start_date: date | None = None
    due_date: date | None = None
    status: JobStatus | None = None
    notes: str | None = None


class JobResponse(JobBase):
    """API response for job entity."""

    id: int
    customer_name: str | None = None
    assignee_names: list[str] = Field(default_factory=list)
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
